src/Microsoft.DotNet.Interactive.HttpRequestParser/HttpRequestParser.cs

```diff
@@ -39,24 +41,151 @@ public HttpSyntaxTree Parse()
         {
             _tokens = new HttpLexer(_sourceText, _syntaxTree).Lex();
 
+            var unmatchedComments = new List<HttpCommentNode>();
```

Comment:
I'd probably just call this `comments`. They're matched, they're just not parented yet. Or maybe something like `commentsToPrepend`?